frontend/src/components/MapFilters/PublishedHiddenFilter/PublishedHiddenFilter.tsx

```diff
@@ -0,0 +1,160 @@
+        onChange(status, since, dateStr);
+        console.log(dateStr, dateOk);
+    };
+
+    const isSelected = (value: string): boolean => isPublic === value;
+
+    return (
+        <div>
+            <MapFilterItem header="Julkaistu/piilotettu" eventKey={eventKey}>
+                <div className="published-hidden-filters">
+                    {"Kätkö on"}
+                    <div className="radio-button-filter">
+                        <input
+                            id="julkaistu"
+                            type="radio"
+                            name="julkaistu"
+                            value="julkaistu"
+                            checked={isSelected("julkaistu") || DEFAULT_DISPLAY_VALUE}
+                            onChange={modifyIsPublic}
+                            
+                        />
+                        <label htmlFor="julkaistu">julkaistu</label>
+                    </div>
+                    <div className="radio-button-filter">
+                        <input
+                            id="piilotettu"
+                            type="radio"
+                            name="piilotettu"
+                            value="piilotettu"
+                            checked={isSelected("piilotettu") || DEFAULT_DISPLAY_VALUE}
+                            onChange={modifyIsPublic}
+                        />
+                        <label htmlFor="piilotettu">piilotettu</label>
+                    </div>
+                    <div className="radio-button-filter">
+                        <input
+                            id="julkaistuTaiPiilotettu"
+                            type="radio"
+                            name="julkaistuTaiPiilotettu"
+                            value="julkaistuTaiPiilotettu"
+                            checked={isSelected("julkaistuTaiPiilotettu") || DEFAULT_DISPLAY_VALUE}
+                            onChange={modifyIsPublic}
+                        />
+                        <label htmlFor="julkaistuTaiPiilotettu">julkaistu tai piilotettu</label>
+                    </div>
+
+                    <div className="public-text-input">
+                        {"Mistä: "}
+                        <input
+                            name="publicSince"
+                            type="text"
+                            id="publicSince"
+                            onChange={modifyPublicSince}
+                            value={publicSince}
+                            placeholder="dd.mm.yyyy"
+                            data-testid="public-since-input"></input>
+                    </div>
+
+                    <div className="public-text-input">
+                        {"Mihin: "}
+                        <input
+                            name="publicUntil"
+                            type="text"
+                            id="publicUntil"
+                            onChange={modifyPublicUntil}
+                            value={publicUntil}
```

Comment:
The issue where the value is not updated correctly after reset can be solved by giving it an empty string if publicUntil is undefined. Like this:

value={publicUntil || ""}

Reply:
Text input content now disappears when clicking "Tyhjennä" (Reset), but only when closing and opening filters menu again.

Reply:
This has been changed